### PR TITLE
Language support for CodeQL enablement

### PR DESCRIPTION
**Pull request by coveros-phil**
Added functionality to support automatically detecting the language o…f a repo when enabling codeQL. This allows for selecting a specific yml file based on the primary language of the repo. Also if there is no primary language or a language not supported by codeQL it will not enable code scanning

### CODE REVIEW

File: bin/workflows/codeql-analysis-csharp.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      # Initializes the CodeQL tools for scanning.
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          languages: ${{ matrix.language }}
+          queries: security-extended, security-and-quality
```
Comment (NickLiffen) [inline bin/workflows/codeql-analysis-csharp.yml:42]:
```suggestion
          queries: security-extended
```

File: bin/workflows/codeql-analysis-cpp.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      - name: Autobuild
+        uses: github/codeql-action/autobuild@v2
+
+      - name: Perform CodeQL Analysis
+        uses: github/codeql-action/analyze@v2
+        with:
+        # If you would like to have the scanning results uploaded as a SARIF artifcat uncomment the below lines
+        # output: code-scanning-results
+
+      # - name: Upload SARIF as a Build Artifact
```
Comment (NickLiffen) [inline bin/workflows/codeql-analysis-cpp.yml:53]: I would un-comment this 👍  we do want to do the upload 👍

File: bin/workflows/codeql-analysis-csharp.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      - name: Autobuild
+        uses: github/codeql-action/autobuild@v2
+
+      - name: Perform CodeQL Analysis
+        uses: github/codeql-action/analyze@v2
+        with:
+        # If you would like to have the scanning results uploaded as a SARIF artifcat uncomment the below lines
+        # output: code-scanning-results
+
+      # - name: Upload SARIF as a Build Artifact
```
Comment (NickLiffen) [inline bin/workflows/codeql-analysis-csharp.yml:53]: I would un-comment this 👍  we do want to do the upload 👍

File: bin/workflows/codeql-analysis-cpp.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      # Initializes the CodeQL tools for scanning.
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          languages: ${{ matrix.language }}
+          queries: security-extended, security-and-quality
```
Comment (NickLiffen) [inline bin/workflows/codeql-analysis-cpp.yml:42]:
```suggestion
          queries: security-extended
```

File: bin/workflows/codeql-analysis-go.yml [hunk 5]
```diff
@@ -41,27 +39,17 @@ jobs:
...
-      # 📚 https://git.io/JvXDl
-
-      # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
-      #    and modify them (or add more) to build your code if your project
-      #    uses a compiled language
-
-      #- run: |
-      #   make bootstrap
-      #   make release
+          queries: security-extended, security-and-quality
```
Comment (NickLiffen) [inline bin/workflows/codeql-analysis-go.yml:42]:
```suggestion
          queries: security-extended
```

**Comment by NickLiffen** [comment]
The code looks 👍 to me.

@coveros-phil, can you quickly post some screenshots showing this working, please 🙇 E.G just the terminal output of getting the repos and then enabling it 👍

**Comment by coveros-phil** [comment]
No problem, here you go. If you need more stuff just lmk and I can get more screenshots.

Screen Shot 2022-11-10 at 1 32 47 PM
Screen Shot 2022-11-10 at 1 34 07 PM
Screen Shot 2022-11-10 at 1 36 52 PM

**Comment by NickLiffen** [comment]
I have no idea why that status check is failing 😢 But this PR is 👍 Thanks SO much for baring with me in getting this over the line 👍

**Pull request merged**